File: uniapp-web/src/api/menu.ts
/**
 * 菜单管理接口
 */
import { get, post, put, del } from './request'

export interface MenuItem {
  id: number
  name: string
  icon: string
  path: string
  parent_id: number
  sort_order: number
  status: number
  created_at?: string
  updated_at?: string
  children?: MenuItem[]
}

/** 获取菜单列表（树形结构） */
export function getMenuTree() {
  return get<MenuItem[]>('/api/menus?tree=true')
}

/** 获取菜单平铺列表 */
export function getMenuList() {
  return get<MenuItem[]>('/api/menus?tree=false')
}

/** 获取单个菜单详情 */
export function getMenuDetail(id: number) {
  return get<MenuItem>(`/api/menus/${id}`)
}

/** 新增菜单 */
export function addMenu(data: Partial<MenuItem>) {
  return post<MenuItem>('/api/menus', data)
}

/** 修改菜单 */
export function updateMenu(id: number, data: Partial<MenuItem>) {
  return put<MenuItem>(`/api/menus/${id}`, data)
}

/** 删除菜单 */
export function deleteMenu(id: number) {
  return del(`/api/menus/${id}`)
}

File: uniapp-server/schemas/menu.py
"""菜单相关数据模型"""
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class MenuCreate(BaseModel):
    """创建菜单请求"""
    name: str
    icon: Optional[str] = ""
    path: Optional[str] = ""
    parent_id: Optional[int] = 0
    sort_order: Optional[int] = 0
    status: Optional[int] = 1


class MenuUpdate(BaseModel):
    """更新菜单请求"""
    name: Optional[str] = None
    icon: Optional[str] = None
    path: Optional[str] = None
    parent_id: Optional[int] = None
    sort_order: Optional[int] = None
    status: Optional[int] = None


class MenuInfo(BaseModel):
    """菜单信息响应"""
    id: int
    name: str
    icon: str
    path: str
    parent_id: int
    sort_order: int
    status: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    children: Optional[List["MenuInfo"]] = None

    class Config:
        from_attributes = True